### PR TITLE
Port MakeLocalFunctionStatic analyzer and tests to shared layer

### PR DESCRIPTION
**Pull request by mavasani**
First commit: File moves
Second commit: xlf and resx changes
Third commit: Revert some unintentional changes to fixer

### CODE REVIEW

File: src/Analyzers/CSharp/Analyzers/MakeLocalFunctionStatic/MakeLocalFunctionStaticHelper.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.CSharp.MakeLocalFunctionStatic
+{
+    internal static class MakeLocalFunctionStaticHelper
```
Comment (mavasani) [inline src/Analyzers/CSharp/Analyzers/MakeLocalFunctionStatic/MakeLocalFunctionStaticHelper.cs:15]: Existing code just split into separate helper types for analyzer and fixer.

File: src/Features/CSharp/Portable/MakeLocalFunctionStatic/MakeLocalFunctionStaticCodeFixHelper.cs [hunk 2]
```diff
@@ -155,7 +133,7 @@ private static bool CanMakeLocalFunctionStatic(ImmutableArray<ISymbol> captures)
                         {
                             syntaxEditor.ReplaceNode(
                                 identifierNode,
-                                (node, generator) => generator.IdentifierName(parameter.Name.ToIdentifierToken()).WithTriviaFrom(node));
```
Comment (mavasani) [inline src/Features/CSharp/Portable/MakeLocalFunctionStatic/MakeLocalFunctionStaticCodeFixHelper.cs]: Reverted this change in last commit.

File: src/Features/CSharp/Portable/MakeLocalFunctionStatic/MakeLocalFunctionStaticCodeFixHelper.cs [hunk 3]
```diff
@@ -173,11 +151,11 @@ private static bool CanMakeLocalFunctionStatic(ImmutableArray<ISymbol> captures)
 
                     if (shouldWarn)
                     {
-                        var annotation = WarningAnnotation.Create(CSharpFeaturesResources.Warning_colon_Adding_parameters_to_local_function_declaration_may_produce_invalid_code);
+                        var annotation = WarningAnnotation.Create(CSharpCodeFixesResources.Warning_colon_Adding_parameters_to_local_function_declaration_may_produce_invalid_code);
                         localFunctionWithNewParameters = localFunctionWithNewParameters.WithAdditionalAnnotations(annotation);
                     }
 
-                    return AddStaticModifier(localFunctionWithNewParameters, CSharpSyntaxGenerator.Instance);
+                    return AddStaticModifier(localFunctionWithNewParameters, SyntaxGenerator.GetGenerator(document));
```
Comment (mavasani) [inline src/Features/CSharp/Portable/MakeLocalFunctionStatic/MakeLocalFunctionStaticCodeFixHelper.cs]: Reverted this change in last commit.

File: src/Features/CSharp/Portable/MakeLocalFunctionStatic/MakeLocalFunctionStaticCodeFixHelper.cs (hunk 2 above)
Comment (CyrusNajmabadi) [reply src/Features/CSharp/Portable/MakeLocalFunctionStatic/MakeLocalFunctionStaticCodeFixHelper.cs]: unfortunate if they have different behavior.  i want your change here to be good :)

Comment (mavasani) [reply src/Features/CSharp/Portable/MakeLocalFunctionStatic/MakeLocalFunctionStaticCodeFixHelper.cs]: Tests passed either way, just reverted as wasn't necessary.

**Pull request merged**